src/aleph/vm/controllers/qemu/snapshot_manager.py

```diff
@@ -0,0 +1,121 @@
+import asyncio
+import logging
+import threading
+from time import sleep
+
+from aleph_message.models import ItemHash
+from schedule import Job, Scheduler
+
+from aleph.vm.conf import settings
+from aleph.vm.controllers.firecracker.snapshots import CompressedDiskVolumeSnapshot
+
+logger = logging.getLogger(__name__)
+
+
+def wrap_async_snapshot(vm):
+    asyncio.run(do_vm_snapshot(vm))
+
+
+def run_threaded_snapshot(vm):
+    job_thread = threading.Thread(target=wrap_async_snapshot, args=(vm,))
+    job_thread.start()
+
+
+async def do_vm_snapshot(vm) -> CompressedDiskVolumeSnapshot:
+    try:
+        logger.debug(f"Starting new snapshot for QEMU VM {vm.vm_hash}")
+        assert vm, "VM execution not set"
+
+        snapshot = await vm.create_snapshot()
+        logger.debug(f"New snapshot for QEMU VM {vm.vm_hash} created successfully")
+        return snapshot
+    except ValueError as error:
+        msg = "Failed to create QEMU VM snapshot"
+        raise ValueError(msg) from error
+
+
+def infinite_run_scheduler_jobs(scheduler: Scheduler) -> None:
+    while True:
+        scheduler.run_pending()
+        sleep(1)
+
+
+class QemuSnapshotExecution:
+    vm_hash: ItemHash
+    execution: any  # AlephQemuInstance
+    frequency: int
+    _scheduler: Scheduler
+    _job: Job
+
+    def __init__(
+        self,
+        scheduler: Scheduler,
+        vm_hash: ItemHash,
+        execution,
+        frequency: int,
+    ):
+        self.vm_hash = vm_hash
+        self.execution = execution
+        self.frequency = frequency
+        self._scheduler = scheduler
+
+    async def start(self) -> None:
+        logger.debug(f"Starting QEMU snapshots for VM {self.vm_hash} every {self.frequency} minutes")
+        job = self._scheduler.every(self.frequency).minutes.do(run_threaded_snapshot, self.execution)
```

Comment:
```suggestion
        self._job = self._scheduler.every(self.frequency).minutes.do(run_threaded_snapshot, self.execution)
```

And remove the next line